combustion-android-ble/src/main/java/inc/combustion/framework/ble/ProbeManager.kt

```diff
@@ -488,6 +489,112 @@ internal open class ProbeManager (
             }
     }
 
+    private fun update(): Probe {
+        _probe = _probe.copy(
+                serialNumber = advertisingData.serialNumber,
+                mac = advertisingData.mac,
+                sessionInfo = sessionInfo,
+                minSequenceNumber = probeStatus?.minSequenceNumber ?: _probe.minSequenceNumber,
+                maxSequenceNumber = probeStatus?.maxSequenceNumber ?: _probe.maxSequenceNumber,
+                rssi = if(isConnected.get()) remoteRssi.get() else advertisingData.rssi,
+                id = probeStatus?.id ?: advertisingData.id,
+                color = probeStatus?.color ?: advertisingData.color,
+                batteryStatus = probeStatus?.batteryStatus ?: advertisingData.batteryStatus,
+                connectionState = connectionState,
+                uploadState = uploadState
+        )
+
+        val temperatures = probeStatus?.temperatures ?: advertisingData.probeTemperatures
+        val mode = probeStatus?.mode ?: advertisingData.mode
+
+        if(mode == ProbeMode.INSTANT_READ) {
+            instantReadMonitor.activity()
+
+            _probe = _probe.copy(
+                instantReadCelsius = temperatures.values[0]
+            )
+        }
+        else if(mode == ProbeMode.NORMAL) {
+            val virtualSensors = probeStatus?.virtualSensors ?: advertisingData.virtualSensors
+            val instantReadCelsius = if(!instantReadMonitor.isIdle(PROBE_INSTANT_READ_IDLE_TIMEOUT_MS)) _probe.instantReadCelsius else null
+
+            predictionStatusMonitor.activity()
+
+            predictionStatus = probeStatus?.predictionStatus
+
+            // handle large predictions and prediction resolution
+            if(predictionStatus != null) {
```

Comment:
Nitpick: I think it would be safer to use let:
`predictionStatus?.let {
    // use the prediction status
}`

Reply:
good call.  got it.